File: TransCoder_tokenized_test_set_functions/python/FIND_INDEX_MAXIMUM_OCCURRING_ELEMENT_EQUAL_PROBABILITY.py
def findRandomIndexOfMax ( arr, n ) :
    mp = dict ( )
    for i in range ( n ) :
        if ( arr [ i ] in mp ) :
            mp [ arr [ i ] ] = mp [ arr [ i ] ] + 1
        else :
            mp [ arr [ i ] ] = 1
    max_element = - 323567
    max_so_far = - 323567
    for p in mp :
        if ( mp [ p ] > max_so_far ) :
            max_so_far = mp [ p ]
            max_element = p
    r = int ( ( ( random . randrange ( 1, max_so_far, 2 ) % max_so_far ) + 1 ) )
    i = 0
    count = 0
    while ( i < n ) :
        if ( arr [ i ] == max_element ) :
            count = count + 1
        if ( count == r ) :
            print ( "Element with maximum frequency present at index ", i )
            break
        i = i + 1
    

File: TransCoder_tokenized_test_set_functions/java/FIND_INDEX_MAXIMUM_OCCURRING_ELEMENT_EQUAL_PROBABILITY.java
static void findRandomIndexOfMax ( int arr [ ], int n ) {
  HashMap < Integer, Integer > mp = new HashMap < Integer, Integer > ( );
  for ( int i = 0;
  i < n;
  i ++ ) if ( mp . containsKey ( arr [ i ] ) ) {
    mp . put ( arr [ i ], mp . get ( arr [ i ] ) + 1 );
  }
  else {
    mp . put ( arr [ i ], 1 );
  }
  int max_element = Integer . MIN_VALUE;
  int max_so_far = Integer . MIN_VALUE;
  for ( Map . Entry < Integer, Integer > p : mp . entrySet ( ) ) {
    if ( p . getValue ( ) > max_so_far ) {
      max_so_far = p . getValue ( );
      max_element = p . getKey ( );
    }
  }
  int r = ( int ) ( ( new Random ( ) . nextInt ( max_so_far ) % max_so_far ) + 1 );
  for ( int i = 0, count = 0;
  i < n;
  i ++ ) {
    if ( arr [ i ] == max_element ) count ++;
    if ( count == r ) {
      System . out . print ( "Element with maximum frequency present " + "at index " + i + "\n" );
      break;
    }
  }
}